src/cascade/executor/no_covariate_main.py

```diff
@@ -158,95 +169,57 @@ def age_year_from_data(df):
-        "time_difference_prior": "uniform",
-        "const_value": constraint["mean"],
-    })
-
-
-def internal_model(model_context, inputs):
-    config = model_context.parameters
-    model = model_context.input_data
+    ages = constraint["age_start"].unique().tolist()
+    times = constraint["year_start"].unique().tolist()
+    grid = AgeTimeGrid(ages, times)
+    smoothing_prior = PriorGrid(grid)
+    smoothing_prior[:, :].prior = UniformPrior(-np.inf, np.inf, 0)
+
+    value_prior = PriorGrid(grid)
+    # TODO: change the PriorGrid API to handle this elegantly
+    for _, row in constraint.iterrows():
+        value_prior[row["age_start"], row["year_start"]].prior = ConstantPrior(row["mean"])
+    return Smooth(smoothing_prior, smoothing_prior, value_prior)
+
+
+def internal_model(model, inputs):
+    config = model.parameters
     # convert the observations to a normalized format.
-    model.observations = bundle_to_observations(config, inputs.observations)
-    model.constraints = bundle_to_observations(config, inputs.constraints)
-
-    rates_to_calculate_str = config.non_zero_rates.split()
-    age_df, time_df = age_year_from_data(inputs.constraints)
-    desired_outputs = integrand_outputs(
-        rates_to_calculate_str + ["prevalence"],
-        config.location_id, age_df, time_df)
-    model.outputs = desired_outputs
-
-    model.priors = pd.DataFrame({
-        "prior_name": ["uniform01", "uniform"],
-        "density_id": 0,  # uniform
-        "lower": [1e-10, np.NaN],
-        "upper": [1.0, np.NaN],
-        "mean": [0.01, 0.0],
-        "std": np.array([np.NaN, np.NaN], dtype=np.float),
-        "eta": np.array([np.NaN, np.NaN], dtype=np.float),
-        "nu": np.array([np.NaN, np.NaN], dtype=np.float),
-    })
-
-    smoothing_default = build_smoothing_grid(age_df, time_df)
+    model.input_data.observations = bundle_to_observations(config, inputs.observations)
+    model.input_data.constraints = bundle_to_observations(config, inputs.constraints)
+
+    grid = age_year_from_data(inputs.constraints)
+
+    priors = PriorGrid(grid)
+    priors[:, :].prior = UniformPrior(1e-10, 1.0, 0.01)
+    default_smoothing = Smooth(priors, priors, priors)
+
     # No smoothing for initial prevalence in Brad's example.
-    model.smoothers = {
-        "iota": smoothing_default,
-        "rho": smoothing_default,
-        "chi": smoothing_default,
-        "omega": build_constraint(model.constraints),
-    }
-    LOGGER.debug(f"Omega constraint {model.smoothers['omega']}")
-    return model_context
+    for rate in config.non_zero_rates.split():
+        getattr(model.rates, rate).parent_smooth = default_smoothing
+
+    for rate in config.non_zero_rates.split() + ["prevalence"]:
+        integrand = getattr(model.outputs.integrands, RATE_TO_INTEGRAND[rate].name)
+        integrand.active = True
+        integrand.age_lower = min(default_smoothing.grid.ages)
+        integrand.age_upper = max(default_smoothing.grid.ages)
```

Comment:
I'm not sure these lines are right. Seems to be setting a min/max age for the output integrand.  Ohhh... this is for avgint table. Got it. That documentation is unclear, and I think this code isn't correct. The min/max isn't for the lowest possible. Each entry in the avgint has its own min and max, like 25-30 yr-olds.

Reply:
Oh... Ok. I'll clean this up.